app/src/main/java/bg/zahov/app/ui/authentication/login/LoginFragment.kt

```diff
@@ -0,0 +1,36 @@
+package bg.zahov.app.ui.authentication.login
+
+import android.os.Bundle
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import androidx.compose.ui.platform.ComposeView
+import androidx.compose.ui.platform.ViewCompositionStrategy
+import androidx.fragment.app.Fragment
+import androidx.fragment.app.viewModels
+import androidx.navigation.fragment.findNavController
+import bg.zahov.app.hideBottomNav
+import bg.zahov.app.hideTopBar
+
+class LoginFragment : Fragment() {
+    private val loginViewModel: LoginViewModel by viewModels()
+
+    override fun onCreateView(
+        inflater: LayoutInflater,
+        container: ViewGroup?,
+        savedInstanceState: Bundle?,
+    ): View {
+        return ComposeView(requireContext()).apply {
+            setViewCompositionStrategy(ViewCompositionStrategy.DisposeOnViewTreeLifecycleDestroyed)
+            setContent {
+                LoginScreen(loginViewModel, findNavController())
```

Comment:
The login screen itself doesn't need to hold control over the navController. Use a lambda as an argument callback instead and call the navigation function from this fragment.